run.ps1

```diff
@@ -0,0 +1,30 @@
+# set the base tools directory
+$toolsLocalPath = Join-Path $PSScriptRoot ".tools"
+$bootStrapperPath = Join-Path $toolsLocalPath "bootstrap.ps1"
+
+# if the boot-strapper script doesn't exist then download it
+if ((Test-Path $bootStrapperPath) -eq 0)
+{
+    if ((Test-Path $toolsLocalPath) -eq 0)
+    {
+        mkdir $toolsLocalPath | Out-Null
+    }
+
+    # download boot-strapper script
+    Invoke-WebRequest "https://raw.githubusercontent.com/dotnet/buildtools/master/bootstrap/bootstrap.ps1" -OutFile $bootStrapperPath
+}
+
+# now execute it
+Invoke-Expression "$bootStrapperPath (Get-Location) $toolsLocalPath" | Out-File (Join-Path (Get-Location) "bootstrap.log")
+if ($LastExitCode -ne 0)
+{
+    Write-Output "Boot-strapping failed with exit code $LastExitCode"
+    exit $LastExitCode
+}
+
+# execute the tool using the dotnet.exe host
+$dotNetExe = Join-Path $toolsLocalPath "dotnetcli\shared\Microsoft.NETCore.App\version\dotnet.exe"
```

Comment:
Sorry bringing forward some other comments from the PR you closed.

I still don't understand why you don't need to set this path in the targets? We will want the targets to point at this same dotnet.exe otherwise all the tools will need to agree on the CLI version again which is what we were trying to avoid/workaround here by directly calling dotnet.exe in the version folder.